### PR TITLE
Moved js/*_test.ts to tests/

### PR DESCRIPTION
**Pull request by fxnoob**
#794 
Move js/*_test.ts to tests/

### CODE REVIEW

**Comment by fxnoob** [comment]
Moved js/*_test.ts to tests/ Including unit_test.ts js/testing/ and test_util.ts.

**Comment by kevinkassimo** [comment]
@fxnoob You might want to update `tools/unit_tests.py`, since now the paths to `unit_tests.ts` have changed

https://github.com/denoland/deno/blob/1729bdb0d7f504436cacc07bedba642591fdb0cb/tools/unit_tests.py#L12-L31

**Comment by kevinkassimo** [comment]
Also, for tests such as `console_test.ts`, you might need to update where things are important, for example, `stringifyArgs` is imported directly from `console.ts`, thus the import path also needs update.

**Comment by kevinkassimo** [comment]
Also, to validate if things are still working after move, you can run `./tools/test.py` locally.

File: website/app_test.js [hunk 1]
```diff
@@ -84,16 +76,6 @@ test(function createBinarySizeColumnsIrregularData() {
   assertEqual(columns, [["binary_size", 0, 0]]);
 });
```
Comment (kevinkassimo) [inline website/app_test.js]: Please don't delete tests from master

Comment (fxnoob) [reply website/app_test.js]: I've moved to tests in another directory. i've not deleted them, i guess?

Comment (kevinkassimo) [reply website/app_test.js]: I mean some tests currently in masters `website/app_test.js`, I don't think they are moved to somewhere else. Looks like you rebased on master but probably unexpectedly dropped these test cases during the process

Comment (fxnoob) [reply website/app_test.js]: when I had to resolve the conflict in website/app_test.js,I did not resolve it properly and it caused dropping these test cases during the process. My bad.would you please help me in resolving the conflicts?

**Comment by piscisaureus** [comment]
Please also update or remove this "ignore list" of files that don't need to be bundled: https://github.com/denoland/deno/blob/c226e1b44e9fc4f2cb77b6eea5d0464186ba66a8/.appveyor.yml#L392

**Comment by ry** [comment]
FYI I think ignore should be empty after this update
https://github.com/denoland/deno/blob/c226e1b44e9fc4f2cb77b6eea5d0464186ba66a8/.appveyor.yml#L392

**Comment by fxnoob** [comment]
> Please also update or remove this "ignore list" of files that don't need to be bundled:
> 
> [deno/.appveyor.yml](https://github.com/denoland/deno/blob/c226e1b44e9fc4f2cb77b6eea5d0464186ba66a8/.appveyor.yml#L392)
> 
> Line 392 in [c226e1b](/denoland/deno/commit/c226e1b44e9fc4f2cb77b6eea5d0464186ba66a8)
> 
>        $ignore = "test_util.ts", "unit_tests.ts", "*_test.ts"

Sure.

**Comment by ry** [comment]
@fxnoob I'm very sorry about this but I've got to close this PR. I really appreciate the effort - and although I'm not merging this - I've been watching and it has helped me think about the unit tests. To be honest we're not ready to make this change: [some of our unit tests are against internal interfaces](https://github.com/denoland/deno/pull/819), most are against the public interface.
Thank you.

**Pull request closed without merging**